fix1531 add UI setting of harvester-monitoring (#124)
harvester/docs: docs/monitoring/monitoring.md

## VM Detail Metrics
For each VM, users can view the VM metrics by clicking the VM details page.
![](./assets/vm-metrics.png)

## Change resources setting of Monitoring
_Available as of v1.0.1_

> Monitoring has a couple of components to collect metrics data from all NODEs/PODs/VMs/... and summarize them. The resources required by Monitoring are related to workloads and hardware resources. Harvester sets the default values according to engineering practices, users can change them accordingly.

Reviewer: Is there any reference for the `engineering practices` mentioned?

Author: I do not have an reference data/article, it is more from an roughly estimation based on daily work and experience.

> Monitoring has a couple of components to collect metrics data from all NODEs/PODs/VMs/... and summarize them. The resources required by Monitoring are related to workloads and hardware resources. Harvester sets the default values according to engineering practices, users can change them accordingly.

Reviewer: ```suggestion
The `Monitoring` has a couple of components to collect metrics data from all NODEs/PODs/VMs/... and summarize them. The resources required by the `Monitoring` are related to workloads and hardware resources. Harvester sets the default values according to engineering practices, and you can change them accordingly.
```

Author: updated, thanks.

> The resources setting of two of the components are available

Reviewer: ```suggestion
The following two components `resources settings` are available:
```

(1) Monitoring-Prometheus

(2) Monitoring-Prometheus-node-exporter

> ### Change resources setting of Monitoring-Prometheus

Reviewer: ```suggestion
### Change resources settings of Monitoring-Prometheus
```

> In `Advanced` `Settings` page, user can view and change the resources setting of Monitoring.

Reviewer: ```suggestion
In the `Advanced Settings` page, you can view and change the resources settings as follow:
```

> (1) Naviate to settings page, find `harvester monitoring`

Reviewer: ```suggestion
(1) Navigate to the settings page, find `harvester monitoring`.
```

![](./assets/monitoring-setting.png)

> (2) Click `Show harvester-monitoring` to view current value of setting

Reviewer: ```suggestion
(2) Click `Show harvester-monitoring` to view the current values.
```

![](./assets/monitoring-setting-view-current.png)

> (3) Click up-right corner pop up menu and select `Edit Setting` to set new value

Reviewer: ```suggestion
(3) Click in the up-right corner pop up menu and select `Edit Setting` to set a new value.
```

![](./assets/monitoring-setting-edit-config.png)

> (4) Click `Save`, wait a while, the monitoring will be restarted with new resources setting

Reviewer: ```suggestion
(4) Click `Save`. The `Monitoring` will be restarted with the new resources settings. Please note, the restart can take some time.
```

The most frequently used option is memory setting.

> `Requested Memory` is the minimum memory of the `Monitoring`, the recommended value is about 5% to 10% of system memory of one single management node. A value of less than 500Mi will be denied. 

Reviewer: ```suggestion
`Requested Memory` is the minimum memory of the `Monitoring`. The recommended value is about 5% to 10% of system memory of one single management node. A value less than 500Mi will be denied. 
```

> `Memory Limit` is the maximum memory of the `Monitoring`, the recommended value is about 30% of system memory of one single management node, when monitoring touches this value, it will be restarted.

Reviewer: ```suggestion
`Memory Limit` is the maximum memory of the `Monitoring`. The recommended value is about 30% of system memory of one single management node, when the `Monitoring` reaches this value, it will be restarted.
```

> Depends on the available hardware resources and system loads, user may change the setting accordingly.

Reviewer: ```suggestion
Depending on the available hardware resources and system loads, you may change the settings accordingly.
```

> Notice: when the (3) management node(s) have different hardware resources, please set the value based on the fewest one.

Reviewer: ```suggestion
!!! note
    If you have multiple management nodes with different hardware resources, please set the value based on the smaller one.
```

> ### Change resources setting of Monitoring-Prometheus-node-exporter

Reviewer: ```suggestion
### Change resources settings of Monitoring-Prometheus-node-exporter
```

> Monitoring-Prometheus-node-exporter has similar resources specifications as Monitoring-Prometheus, WebUI setting will come in future versions.

Reviewer: ```suggestion
`Monitoring-Prometheus-node-exporter` has a similar resources specifications as `Monitoring-Prometheus`. In future versions, a new `WebUI` setting will be available.
```

> To change the values, users may use CLI command: `kubectl edit managedchart rancher-monitoring -n fleet-local`.

Reviewer: ```suggestion
To update the values, you may use the CLI command: `kubectl edit managedchart rancher-monitoring -n fleet-local`.
```

The related path and default value are:

 `spec.values.prometheus.prometheus-node-exporter.resources.limits.cpu`:`200m`

 `spec.values.prometheus.prometheus-node-exporter.resources.limits.memory`:`180Mi`

 `spec.values.prometheus.prometheus-node-exporter.resources.requests.cpu`:`100m`

 `spec.values.prometheus.prometheus-node-exporter.resources.requests.memory`:`30Mi`

> When many VMs are deployed in one NODE, the OOM(out of memory)/abnormal restarting of prometheus-node-exporter POD(s) may be observed, then users can change the `limits.memory` to a bigger value.

Reviewer: ```suggestion
!!! attention
    When many VMs are deployed in one Node, the OOM (out of memory)/abnormal restarting of `prometheus-node-exporter` Pod(s) may be observed. In that case, you should change the `limits.memory` to a bigger value.
```
